hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/cache/ConcurrentHashMapLruCache.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.jet.sql.impl.cache;
+
+import javax.annotation.concurrent.GuardedBy;
+import java.util.Deque;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.function.Function;
+
+/**
+ * Implementation of a simple LRU cache based on {@link ConcurrentHashMap}. It is faster than Guava cache when
+ * there are a lot of reads and few writes.
+ */
+public class ConcurrentHashMapLruCache<K, V> implements Cache<K, V> {
```

Comment:
I don't think we should keep unused classes in the code, unless we already foresee that it will be actually used.

Reply:
Done